### ISSUE TITLE
how can i change the feature importance type using a pipeline object?

### ISSUE THREAD

**Issue by zahs123** [dmlc/xgboost#6370]
i am trying to compare xgboost and light gradient boosting feature importance but both have different assessment criteria. so i am trying to change the feature importance type and see how i can do this. 

e.g. 

```
pipeline = make_pipeline(XGBClassifier())
param_grid =    { 
                  'xgbclassifier__learning_rate': [0.01,0.005,0.001],
                 
                  }

gini_scorer = make_scorer(normalized_gini, greater_is_better = True)

# Initialize Grid Search Modelg
model = GridSearchCV(pipeline,param_grid = param_grid,scoring = gini_scorer,
                                 verbose= 1,iid= True,
                                     refit = True,cv  = 3)

model.fit(x, y)
print("Best score: %0.3f" % model.best_score_)
print("Best parameters set:")
model = model.best_estimator_


then to get feature importances i: 

feature_importances = model.steps[-1][1].feature_importances_
pd.DataFrame(feature_importances, index=feature_names,
                     columns=['Importance']).sort_values('Importance') \
            .plot(kind='barh', figsize=(15, 25)) 
```

however i am not sure how i can edit this code of mine to allow a change in feature importance e.g. what if i want it by gain etc..

**Comment by trivialfis** [comment]
Maybe `importance_type` parameter can help?

**Comment by zahs123** [comment]
where do i put this?

**Comment by trivialfis** [comment]
The constructor of `XGBClassifier`.

**Comment by zahs123** [comment]
for some reason, when i try plotting the feature importance two ways where: 
`pipeline = make_pipeline(XGBClassifier(importance_type='weight'))`

1)```
 features_xgboost = model.best_estimator_.named_steps["xgbclassifier"].get_booster().get_score()
keys = list(features_xgboost.keys())
values = list(features_xgboost.values())

data = pd.DataFrame(data=values, index=keys, columns=["score"]).sort_values(by = "score", ascending=True)
data.plot(kind='barh')
```

2 ) ```
features = x_train.columns
feature_names = features.tolist()
pd.DataFrame(feature_importances, index=feature_names,
                     columns=['Importance']).sort_values('Importance') \
            .plot(kind='barh', figsize=(15, 25))
```
i get different scales, how can this be? so for 1) i get the scale between 0-300 with different orders of features as well and in 2) it's between 0-0.04

**Comment by RoyalTS** [comment]
`XGBClassifier.feature_importances_` normalizes the feature importances. If you divide the importances in (1) by their sum you should get the same as in (2).
